frontend/src/router/index.js

```diff
@@ -0,0 +1,258 @@
+          meta: {
+            authRequired: true,
+            permissionsRequired: [
+              'View Vehicles'
+            ]
+          },
+          component: Inventory
+        },
+      ]
+    },
+    {
+      path: '/pages',
+      redirect: '/pages/404',
+      name: 'Pages',
+      component: {
+        render(c) { return c('router-view') }
+      },
+      children: [
+        {
+          path: '404',
+          name: 'Page404',
+          component: Page404
+        },
+        {
+          path: '500',
+          name: 'Page500',
+          component: Page500
+        },
+        {
+          path: 'login',
+          name: 'Login',
+          meta: {
+            unAuthRequired: true
+          },
+          component: Login
+        },
+        {
+          path: 'register',
+          name: 'Register',
+          meta: {
+            unAuthRequired: true
+          },
+          component: Register
+        },
+        {
+          path: 'changePassword',
+          name: 'Change Password',
+          meta: {
+            authRequired: true
+          },
+          component: ChangePassword
+        }
+      ]
+    }
+  ]
+});
+
+router.beforeEach((to, from, next) => {
+
+  // if the promptPasswordChange flag is true and we're not already going to the 
+  // change password page, redirect to change password page
+  if (Store.state.auth.promptPasswordChange && to.path != '/pages/changePassword') {
+    next('/pages/changePassword');
+  }
+
+  else if (Store.state.auth.role != "Administration" && to.name == 'Dealerships') {
+    next(`/dealerships/${Store.state.auth.dealership}`);
+  }
+
+  else if (!Store.state.auth.promptPasswordChange && to.path == '/pages/changePassword') {
+    next('/dashboard');
+  }
+
+  // add the meta tag "authRequired: true" to any routes you want protected
+  else if (to.meta.authRequired && !Store.state.auth.token) {
+    next('/pages/login');
+  }
+
+  // add the meta tage "unAuthRequired: true" to any routes which requires the user
+  // to be logged out to access
+  else if (to.meta.unAuthRequired && !!Store.state.auth.token) {
+    next('/dashboard');
+  }
+
+  // add the meta tag "permissionsRequired: [permissions]" to any routes needing specific permissions
+  else if (to.meta.permissionsRequired) {
+    // go through the permissionsRequired list and verify they exist in the logged in user permissions
+    // if not, call next(false) to cancel the request
+
+    const BreakLoop = {};
+    let authorized = true;
+    try {
+      to.meta.permissionsRequired.forEach(permission => {
+        if (!Store.state.auth.userPermissions.includes(permission)) {
+          throw BreakLoop;
+        }
+      });
+    } catch (e) {
+      authorized = false;
+    }
+    // user has the correct permissions, therefore continue with the request
+    if (authorized)
+      next();
+    else
+      next('/dashboard');
+  }
+
+  // all checks validated, therefore continue with the request
+  else {
+    next();
```

Comment:
This segment has too many conditional statements, changing to a switch case will reduce cognitive complexity.

Reply:
A switch statement in this case wouldn't work since it isn't the same variable being logically operated on every time. Unfortunately using `if` statements is the only way to check all the cases before routing.